.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,61 @@
+name: Benchmarking tests
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - ci-*
```

Comment:
Should ideally be done weekly or bi-weekly and only on `main`. For testing, I am running it on PRs.

Reply:
Yup. Will keep it to bi-weekly. Will incorporate that change once we're one with how we want report the benchmarks and the pipelines we want to benchmark.

Reply:
Should change this to a cron job

Reply:
Yes this will be changed.